src/NServiceBus.Core.Tests/Licensing/LicenseManagerTests.cs

```diff
@@ -0,0 +1,279 @@
+﻿namespace NServiceBus.Core.Tests.Licensing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using NServiceBus.Logging;
+    using NUnit.Framework;
+    using Particular.Licensing;
+
+    [TestFixture]
+    public class LicenseManagerTests
+    {
+        [TestCase("2012-12-12", "2012-12-13")]
+        [TestCase("2012-12-12", "2012-12-14")]
+        [TestCase("2012-12-12", "2012-12-15")]
+        public void ShouldWarnAboutExpiringTrialLicense(string currentDate, string expirationDate)
+        {
+            var licenseManager = new LicenseManager();
+            licenseManager.todayUtc = () => DateTime.ParseExact(currentDate, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal);
```

Comment:
Perhaps make this a helper func or some factory that creates the `LicenseManager`?

Reply:
I've changed the LicenseManager ctor to require a datetimeprovider instead. Are you happy with that approach?

Reply:
Was thinking to hide the .ParseExact(...) away with a:

CreateLicenseManager(currentDate)

or similar to make the tests more terse.

But we’re in nitpick territory now

Reply:
I've went with a helper to parse the datetime string as that's something used more often in the tests (also outside creating the licensemanager). The setup code looks quite acceptable to me now (as your proposal would just wrap the function delegate in addition), thoughts?

Reply:
Yup!